groups/prometheus/main.tf

```diff
@@ -0,0 +1,76 @@
+provider "aws" {
+  region = var.region
+}
+
+terraform {
+  backend "s3" {}
+}
+
+module "security-groups" {
+  source           = "./module-security-groups"
+  region           = var.region
+  vpc_id           = local.vpc_id
+  ssh_cidrs        = concat(local.internal_cidrs,local.vpn_cidrs)
+  prometheus_cidrs = concat(local.mgmt_private_subnet_cidrs)
+  tag_environment  = var.tag_environment
+  tag_service      = var.tag_service
+}
+
```

Comment:
Going to this level of modularity in the way it is being done here, is making life harder for yourself with having to code the entire security group module yourself, and then pass outputs from there into other modules. 
There is two options:
1. Instead of it being an entirely separate module, have one module "prometheus", with the different sections like security group as a .tf file underneath that. This saves having to pass outputs around. Every tf file is linked to the same use case so they don't need to be separate modules.
prometheus-terraform
  ├── terraform               - General terraform config
  ├── module-prometheus    - Config specific to ec2 instances
                                                       ├── instance.tf  - Config for creating and editing route53 records
                                                      └── security-group.tf  - Config for creating and editing security groups

2. The best option, and how terraform should be used (and what we should aim for going forwards) is making use of reusable modules to minimise how much code needs to be written. So keep the same structure, but pull the module from the terraform repository instead of coding it yourself. Outputs are generated by the module for you.
It is used like:
`module "security-group" {
  source  = "terraform-aws-modules/security-group/aws"
  version = "1.0.0"
  # insert the required variables here
}`
The security group module is [here](https://registry.terraform.io/modules/terraform-aws-modules/security-group/aws/1.0.0)
Examples are [here](https://github.com/terraform-aws-modules/terraform-aws-security-group/blob/master/examples/complete/main.tf)

Reply:
This is going to be addressed in another story in the next sprint as it'll take a bit of extra work